### ISSUE TITLE
Question: Howto monitor CGI failures?

### ISSUE THREAD

**Issue by toburger** [isaacabraham/azure-fsharp-helpers#2]
I had a great experience with your helpers in the past!

However, I had problems with my service that it couldn't start up the process anymore, because of some strange error when the HttpPlatformHandler tried to restart my [app.](http://stackoverflow.com/questions/41097564/azure-suave-fails-sometimes-with-httpplatfomhandler-and-doesnt-come-back-up)
This failure didn't get detected by AppInsights, because the program never started.

I've seen that there exists a "Application Insights" extension which can be enabled for the App Service.
I wonder if this adds any value to the app and can detect 502 errors.

The downside of activating the extension is, than now every request counts twice.

Do you have any experience or can you give any advice with 502 errors?

**Comment by isaacabraham** [comment]
I actually still don't have a good understand of exactly what that extension does :-) @numberbycolors can you elaborate a bit on what this app service extension actually does :-)

@davidebbo have you ever seen this error (see the SO link in the OP) with HTTP Platform Handler and the App Service getting locked into this "stuck" loop state? I have no clue.

**Comment by NumberByColors** [comment]
@toburger You mean adding Application Insights to your App Service via the Azure Portal? That turns on automatic collection of requests, dependency calls, unhandled exceptions, etc. There's also an Application Insights extension for Visual Studio that adds request/exception CodeLens and telemetry Search: https://marketplace.visualstudio.com/items?itemName=VisualStudioOnlineApplicationInsights.DeveloperAnalyticsTools

**Comment by isaacabraham** [comment]
@NumberByColors Yes, I meant the Azure Portal extension. What exactly does it do - it doesn't hook into code so does it magically hook into the IIS pipeline? Or something else?

**Comment by toburger** [comment]
So, if I understand correctly this extension supersedes the helpers?

**Comment by NumberByColors** [comment]
I _think_ the extension hooks into the IIS pipeline. It doesn't make code changes (it doesn't have the source), but there's a chance there's some special monitoring infrastructure in Azure App Service I'm unaware of. Still asking around to folks on the team who'd know for sure.

**Comment by isaacabraham** [comment]
@toburger I would say supplements rather than supercedes. This will give you IIS level logging I think but for app level logging you'll need the other one still.
